specifyweb/frontend/js_src/lib/components/Toolbar/DataModel.tsx

```diff
@@ -297,11 +311,35 @@ function DataModelRelationships({
   readonly model: SpecifyModel;
 }): JSX.Element {
   const data = React.useMemo(() => getRelationships(model), [model]);
+
+  const [filteredDependentData, setFilteredDependentData] =
+    React.useState(data);
+  const [dependentFilter, setDependentFilter] = React.useState(false);
+
+  React.useEffect(() => {
+    if (dependentFilter === true) {
+      const dependentData = data
+        .filter((table) => table.isDependent === 'Yes')
+        .map((table) => table);
+      setFilteredDependentData(dependentData);
+    } else {
+      setFilteredDependentData(data);
+    }
+  }, [dependentFilter]);
+
   return (
     <>
-      <H3 id={model.name.toLowerCase()}>{schemaText.relationships()}</H3>
+      <div className="flex items-center">
+        <H3 id={model.name.toLowerCase()}>{schemaText.relationships()}</H3>
+        <DependentFilterButton
+          isPressed={dependentFilter}
+          onClick={(): void => setDependentFilter(!dependentFilter)}
+        >
+          {schemaText.showDependent()}
```

Comment:
This name is too long for button label - especially since it would be even longer when localized to other languages.

Also, it looks like your solution does not allow to see independent relationships only.

Consider instead just having two buttons: `Dependent` / `Independent` with a similar toggle logic like in app resources.